### DISCUSSION TITLE
http://localhost/docs/api-reference/upload-file/upload-file

### DISCUSSION THREAD

**Discussion by giscus** [imagekitio/discussions#31] [Docs]
# http://localhost/docs/api-reference/upload-file/upload-file

You can upload files to the ImageKit.io media library from your server-side using private API key authentication.

File size limit:
The maximum upload file size is limited to 25MB on the free plan. On paid plan, this limit is 300MB for video files.

Version limit:
A file can have a maximum of 100 versions.

http://localhost/docs/api-reference/upload-file/upload-file

**Comment by yashaswi-imagekit** [comment]
hiii

**Reply by imagekitio** [reply]
Hey, let me know if you face any problem with the API.
